File: Sorting and Searching/Sum of Three Values/SumofThreeValues.py
import sys
input = sys.stdin.readline

n,x = map(int, input().split())
a = list(map(int,input().split()))
for i in range(n):
    a[i] = (a[i],i+1)
a.sort()

for i in range(n-2):
    l,r = i+1, n-1
    while l<r:
        curr = a[i][0]+a[l][0]+a[r][0]
        if curr==x:
            sys.stdout.write(str(a[i][1])+' '+str(a[l][1])+' '+str(a[r][1])+'\n')
            sys.exit()
        elif curr<x:
            l += 1
        else:
            r -= 1
sys.stdout.write("IMPOSSIBLE\n")

File: Sorting and Searching/Sum of Three Values/SumofThreeValues.cpp
#include<cstdio>
#include<algorithm>

using namespace std;

int main(){
    int n, x;
    scanf("%d %d", &n, &x);
    pair<int, int> a[n];
    for(int i=0;i<n;i++){
        scanf("%d", &a[i].first);
        a[i].second = i+1;
    }
    sort(a, a+n);
    for(int i=0;i<=n-2;i++){
        int l = i+1, r = n-1;
        while(l<r){
            int curr = a[i].first + a[l].first + a[r].first;
            if(curr == x){
                printf("%d %d %d\n", a[i].second, a[l].second, a[r].second);
                return 0;
            }
            else if(curr < x) l++;
            else r--;
        }
    }
    printf("IMPOSSIBLE\n");
}